sda-download/api/sda/sda.go

```diff
@@ -29,6 +33,43 @@ func sanitizeString(str string) string {
 	return pattern.ReplaceAllString(str, "[identifier]: $1")
 }
 
+func reencryptHeader(oldHeader []byte, reencKey string) ([]byte, error) {
+	if config.Config.Reencrypt.Host == "" {
+		return nil, fmt.Errorf("host of the reencrypt service not set")
+	} else if config.Config.Reencrypt.Port == 0 {
+		return nil, fmt.Errorf("port of the reencrypt service not set")
+	}
```

Comment:
Is this to support some kind of "optional" reencryption usage? Otherwise, check during initialisation instead?

Reply:
No, we don't provide "optional" re-encryption usage? I will move this validation to the config. 
One question is when the Host or Port of the reencryption server is not set, but the client public key is provided, should the `sda-download` return the encrypted file or and empty file?

Reply:
Non-short term, I think also the decrypted download case should use the reencryption service, getting rid of the need to have the sensitive key that can decrypt archive contents in the public facing service (e.g. reencrypt can possibly be hosted in an "internal" cluster with only the grpc port exposed), so I believe we will want to require reencryption service to be used (these things configured)

Reply:
We will probably fix this when we merge `download` into `sda`; it will *either* run with reencryption *or* supply decrypted data.

Reply:
Removed in the commit 8dfc45b, and this checking is done in the config.